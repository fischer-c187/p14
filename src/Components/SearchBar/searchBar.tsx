type SearchBarProps = {
  setSearch: (value: string) => void;
};

function SearchBar({ setSearch }: SearchBarProps) {
  return (
    <input
      type='text'
      placeholder='Search...'
      className='bg-glass bg-12-center bg-no-repeat h-10 px-5 pl-11 text-sm bg-white border-2 border-gray-300 rounded-md focus:border-transparent focus:outline-none focus:ring-2 focus:ring-citron-600 group-focus:focus:ring-citron-600'
      onChange={(event) => setSearch(event.target.value)}
    />
  );
}

export default SearchBar;
