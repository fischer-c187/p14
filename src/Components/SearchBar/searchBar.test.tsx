import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import userEvent from "@testing-library/user-event";
import SearchBar from "./searchBar";

describe("SearchBar", () => {
  it("should render without error", () => {
    render(<SearchBar setSearch={() => {}} />);
    expect(screen.getByPlaceholderText("Search...")).toBeInTheDocument();
  });

  it("should call setSearch on change", async () => {
    const setSearch = vi.fn();
    const user = userEvent.setup();
    render(<SearchBar setSearch={setSearch} />);

    await user.type(screen.getByPlaceholderText("Search..."), "test");
    expect(setSearch).toHaveBeenCalledWith("test");
  });

  it("should render expected snapshot", () => {
    const { asFragment } = render(<SearchBar setSearch={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
