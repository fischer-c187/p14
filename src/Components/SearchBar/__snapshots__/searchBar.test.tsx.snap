// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > should render expected snapshot 1`] = `
<DocumentFragment>
  <input
    class="bg-glass bg-12-center bg-no-repeat h-10 px-5 pl-11 text-sm bg-white border-2 border-gray-300 rounded-md focus:border-transparent focus:outline-none focus:ring-2 focus:ring-citron-600 group-focus:focus:ring-citron-600"
    placeholder="Search..."
    type="text"
  />
</DocumentFragment>
`;
